aten/src/ATen/BatchedTensorImpl.cpp

```diff
@@ -61,6 +62,14 @@ int64_t BatchedTensorImpl::actualDim(int64_t dim, bool wrap_dim) const {
   TORCH_INTERNAL_ASSERT(false);
 }
 
+void BatchedTensorImpl::checkInvariants() const {
+  int64_t prev_level = -1;
+  for (const auto& bdim : bdims_) {
+    TORCH_INTERNAL_ASSERT(bdim.level() > prev_level);
```

Comment:
Why not have this return true or false and then use assert at the call-site? That'll make it more reusable in other circumstances.

Reply:
We can always refactor this later to return true or false if there are other call sites that need to handle this in a different way. I don't have any other circumstances in mind for calling this function, do you?

Reply:
It's better as written, because if there are multiple invariants you can give more clear error messages when each one fails.

However, if you want to make this debug only, you're gonna to have some extra ifdef to skip the entire thing.